appcues/src/main/java/com/appcues/analytics/AnalyticsQueueManager.kt

```diff
@@ -0,0 +1,90 @@
+package com.appcues.analytics
+
+import com.appcues.data.remote.request.ActivityRequest
+import com.appcues.data.remote.request.EventRequest
+import java.util.Timer
+import java.util.TimerTask
+import kotlin.concurrent.schedule
+
+internal class AnalyticsQueueManager {
```

Comment:
here I extracted everything related to queue in the analyticsTracker class and made it in its own class. let me know if you would change anything, and also if there is any scenario you remember that you can test this version against to make sure its working the same that would be helpful, but I basically just moved code around to make it cleaner on AnalyticsTracker and also easier to test there. (can comeback to this and also test this class)

Reply:
this seems like a reasonable split to me, to break out some of the responsibilities into its own class for the queue.